app/scripts/models/account.js

```diff
@@ -414,10 +416,13 @@ define(function (require, exports, module) {
         }
       })
       .then((updatedSessionData) => {
+        this._metrics.logEvent('core.signin.success');
```

Comment:
Since the new laptop died, I have to re-write this, grr.

My intuition is this is the wrong place to emit this event. There is some history and some nuance that needs to be out there.

The nuance - `signin.success` is currently emitted in the [signin-mixin.js](https://github.com/mozilla/fxa-content-server/blob/c8b24682b74ad377abc7d75dd8ef24cb6c3decb4/app/scripts/views/mixins/signin-mixin.js#L133:L134). This is because we want to handle two cases:

* Unverified users, or users who must go through signin confirmation are not considered successfully signed in until they complete the email verification loop.
* Users who must view an OAuth permissions screen are not considered successfully signed in until they submit the permissions form.

Emitting the event from the account model handles neither of these cases.

Now for a history lesson about how both `logViewEvent('success')` and `logViewEvent('signin.success')` came into existence.

Many eons ago (uh oh, another one of these) a user could only sign in from the signin screen. We called `this.logViewEvent('success');` after both of the above conditions were met and we could look for `fxa.content.signin.success` in DataDog. All was right with the world.

Next, we added the force_auth screen. Since we were calling `logViewEvent('success');`, to see the total successful signins in DataDog, we could look for `fxa.content.force_auth.success` and `fxa.content.signin.success`. Two events wasn't awesome, but it wasn't bad. There was some confusion about why `fxa.content.signin.success` wasn't the one true metric we should be looking at, but hey, we were young, inexperienced, full of energy, and didn't mind dirty hacks in DataDog to get around our self-created problems.

Later, we decided to add signin from signup, and the metrics world exploded. Since we were only calling `this.logViewEvent('success')`, we had `fxa.content.signin.success`, `fxa.content.force_auth.success`, and NOW, `fxa.content.signup.success`, all for a user that is trying to sign in. Notice in particular the last one. That SAME event is also emitted when the user is signing up from the signup screen. "That won't do" we decided, "how will we know which users are signing in, and which are signing up?" In our naivety, we added a new view event suffix `signin.success`. For a successful signin, we would send to DataDog these three: `fxa.content.signin.signin.success`, `fxa.content.force_auth.signin.success`, and `fxa.content.signup.signin.success`. We could see the rate of successful signins on each page. There was still confusion about what `fxa.content.signin.success` meant, but we ignored that metric because in reality it is an alias of `fxa.content.signin.signin.success`.

Then we decided we wanted our graphs to show us more. We wanted to know the aggregate signin success rate on all the screens. Again, many people asked "what's this fxa.content.signin.success thing, doesn't that give us what we need?" "oh no no, ignore that one. We don't explicitly log what we want, we have to sum these other three metrics."

We did the dirty work in DataDog to sum each of the metrics to find the aggregate success rates and the world moved on, albeit with confusing metrics.

That's the history lesson.

What I think you should do is keep the same `signin.success` semantics as the events that are already emitted. That means moving where the event is logged from the account model out to views/metrics/signin-mixin.js.

My first inclination is to make `fxa.content.signin.success` mean exactly what everyone thinks it means - the user has successfully completed sign in.

In signin-mixin.js, I'd replace [these two lines](https://github.com/mozilla/fxa-content-server/blob/c8b24682b74ad377abc7d75dd8ef24cb6c3decb4/app/scripts/views/mixins/signin-mixin.js#L133:L134):

```js
this.logViewEvent('success');
this.logViewEvent('signin.success');
```

with

```
```js
this.logEvent('signin.success');
this.logViewEvent('signin.success');
```

Replacing `logViewEvent` with `logEvent` will use the event name without a screen name prefix. We'll send `fxa.content.signin.success` to DataDog no matter where the successful signin comes from.

If for whatever reason we *are* using `fxa.content.signin.success` in DataDog (if we are, it should be in very few places), we could create a new suffix, something like:

```js
this.logEvent('signin.complete');
this.logViewEvent('success');
this.logViewEvent('signin.success');
```

That would introduce a new metric, with a new suffix, but it seems a tad less wonky than the `core.` prefix.